nvchecker_source/snapcraft.py

```diff
@@ -0,0 +1,32 @@
+# MIT licensed
+# Copyright (c) 2025 Maxim Slipenko <maxim@slipenko.com>, et al.
+
+from nvchecker.api import (
+  GetVersionError
+)
+
+URL="https://api.snapcraft.io/v2/snaps/info/%(snap)s"
+
+async def get_version(
+  name: str, conf, *,
+  cache, keymanager,
+  **kwargs,
+):
+  try:
+    snap = conf.get("snap")
+    channel = conf.get("channel")
+
+    result = await cache.get_json(
+      URL % { "snap": snap },
+      headers={
+        "Snap-Device-Series": "16",
+      },
+    )
+  except:
+    raise GetVersionError(f"Failed to request snap info for {snap}")
```

Comment:
Please be specific which exception(s) to catch. Use `Exception` if unsure, but please log it so that it's easier to debug.